File: core/src/main/java/info/nightscout/androidaps/db/MedLinkTemporaryBasal.java
package info.nightscout.androidaps.db;

import dagger.android.HasAndroidInjector;

/**
 * Created by Dirceu on 26/05/21.
 */
public class MedLinkTemporaryBasal extends TemporaryBasal{
    private double desiredRate =0d;
    private int desiredPct = 0;

    public MedLinkTemporaryBasal(HasAndroidInjector injector) {
        this.injector = injector;
        injector.androidInjector().inject(this);
    }

    public double getDesiredRate() {
        return desiredRate;
    }

    public void setDesiredRate(double desiredRate) {
        this.desiredRate = desiredRate;
    }

    public int getDesiredPct() {
        return desiredPct;
    }

    public void setDesiredPct(int desiredPct) {
        this.desiredPct = desiredPct;
    }



}

File: database/src/main/java/info/nightscout/androidaps/database/entities/MedLinkTemporaryBasal.kt
package info.nightscout.androidaps.database.entities

import androidx.room.PrimaryKey
import info.nightscout.androidaps.database.embedments.InterfaceIDs
import java.util.*

/**
 * Created by Dirceu on 26/05/21.
 */
// data class MedLinkTemporaryBasal(
//     @PrimaryKey(autoGenerate = true)
//     var id: Long = 0,
//     var version: Int = 0,
//     var dateCreated: Long = -1,
//     var isValid: Boolean = true,
//     var referenceId: Long? = null,
//     var interfaceIDs_backing: InterfaceIDs? = InterfaceIDs(),
//     var timestamp: Long,
//     var utcOffset: Long = TimeZone.getDefault().getOffset(timestamp).toLong(),
//     var type: TemporaryBasal.Type,
//     var isAbsolute: Boolean,
//     var rate: Double,
//     var duration: Long
// )  {
//
//     var percentRate = 0
//     var desiredRate = 0.0
//     var desiredPct = 0
//     fun MedLinkTemporaryBasal.toString(): String {
//         return "MedLinkTemporaryBasal{" +
//             "desiredRate=" + desiredRate +
//             ", desiredPct=" + desiredPct +
//             '}' + super.toString()
//     }
//
// }